debian-setup.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+# Installs all of the necessary packages in a Docker container
+# MUST BE RUN WITH ROOT
+
+apt-get update
+apt-get upgrade -y
```

Comment:
I'm hesitant to keep the upgrade part; [job 156](https://travis-ci.com/ClangBuiltLinux/continuous-integration/jobs/156461095) pulled in an update for tzdata for example.  I assume this will just slow down the tests without any meaningful gain?

Reply:
As I said to Stefan, the upgrade and initial install of packages is no longer than when we were having Travis do it (in fact, it seems a little quicker). Plus, these Debian images are refreshed every month or so I believe so we're not going to updating that much.

Reply:
But you're upgrading packages we don't care about. We probably should not have been doing it with travis.

Removing it does not break anything, right?

Reply:
Fair point I guess; the only thing I am worried about is a default package requiring an upgrade for an installed package but I guess apt will take care of that (I don't know though, I use Arch Linux personally).

I've always been of the mindset that a system should always be up to date but I guess that doesn't matter for these systems since they're only used for one purpose (plus when Debian's image gets refreshed in Docker, we'll get those updates anyways).

Reply:
> I don't know though, I use Arch Linux personally

Dude, that's like a meme on /r/linux. 🐸 

> I've always been of the mindset that a system should always be up to date

I agree for my personal devices.  For a ci testing platform, I don't think we need as strict a requirement.  And it sounds like the base docker image gets updated automatically after a month or so?  So it's never like the existing packages are ever more than a month out of date.

Reply:
> > I don't know though, I use Arch Linux personally
> 
> Dude, that's like a meme on /r/linux. 🐸

Oh believe me, I know :P

> > I've always been of the mindset that a system should always be up to date
> 
> I agree for my personal devices. For a ci testing platform, I don't think we need as strict a requirement. And it sounds like the base docker image gets updated automatically after a month or so? So it's never like the existing packages are ever more than a month out of date.

Hmmm I guess Travis doesn't run `apt-get upgrade`, I thought they did: https://docs.travis-ci.com/user/installing-dependencies/#installing-packages-on-standard-infrastructure

I am not opposed to removing it if you are worried about stuff breaking (although it is Debian stable haha) and fair point regarding the image updates. I just get triggered when I don't have the latest ;P

Will push the removal here in a second.